frontend/app/src/modules/dashboard/history-progress/HistoryQueryIndicator.vue

```diff
@@ -1,16 +1,22 @@
 <script setup lang="ts">
 import type { BlockchainAccount } from '@/types/blockchain/accounts';
 import DateDisplay from '@/components/display/DateDisplay.vue';
+import ChainIcon from '@/components/helper/display/icons/ChainIcon.vue';
+import LocationIcon from '@/components/history/LocationIcon.vue';
 import { useSupportedChains } from '@/composables/info/chains';
 import { useRefWithDebounce } from '@/composables/ref';
 import { useLoggedUserIdentifier } from '@/composables/user/use-logged-user-identifier';
 import { useBlockchainAccountsStore } from '@/modules/accounts/use-blockchain-accounts-store';
+import HashLink from '@/modules/common/links/HashLink.vue';
 import {
   useHistoryQueryIndicatorSettings,
 } from '@/modules/dashboard/history-events/composables/use-history-query-indicator-settings';
+import { useBalanceQueryProgress } from '@/modules/dashboard/history-progress/composables/use-balance-query-progress';
+import { useHistoryQueryProgress } from '@/modules/dashboard/history-progress/composables/use-history-query-progress';
```

Comment:
can we maybe make a unified progress composable and expose all the progress we need from a single place by moving any new code added there?